apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/Indictment.tsx

```diff
@@ -31,16 +36,101 @@ const Indictment: React.FC = () => {
     setWorkingCase,
     isLoadingWorkingCase,
     caseNotFound,
+    isCaseUpToDate,
   } = useContext(FormContext)
   const { formatMessage } = useIntl()
   const { updateCase } = useCase()
+
+  const {
+    createIndictmentCount,
+    updateIndictmentCount,
+    deleteIndictmentCount,
+  } = useIndictmentCounts()
+
   const stepIsValid = true
+
   const handleNavigationTo = useCallback(
     (destination: string) => router.push(`${destination}/${workingCase.id}`),
     [workingCase.id],
   )
+  useDeb(workingCase, ['indictmentIntroduction'])
+
+  const handleCreateIndictmentCount = useCallback(async () => {
+    const indictmentCount = await createIndictmentCount(workingCase.id)
+
+    if (!indictmentCount) {
+      return
+    }
+
+    setWorkingCase((theCase) => ({
+      ...theCase,
+      indictmentCounts: theCase.indictmentCounts
+        ? [...theCase.indictmentCounts, indictmentCount]
+        : [indictmentCount],
+    }))
+  }, [createIndictmentCount, setWorkingCase, workingCase.id])
+
+  useEffect(() => {
+    if (isCaseUpToDate && workingCase.indictmentCounts?.length === 0) {
+      handleCreateIndictmentCount()
+    }
+  }, [
+    isCaseUpToDate,
+    handleCreateIndictmentCount,
+    workingCase.indictmentCounts,
+  ])
+
+  const updateIndictmentCountState = useCallback(
+    (indictmentCountId: string, update: TIndictmentCount) => {
+      setWorkingCase((theCase) => {
+        if (!theCase.indictmentCounts) {
+          return theCase
+        }
 
-  useDeb(workingCase, 'indictmentIntroduction')
+        const indictmentCountIndexToUpdate = theCase.indictmentCounts.findIndex(
+          (indictmentCount) => indictmentCount.id === indictmentCountId,
+        )
+
+        const newIndictmentCounts = [...theCase.indictmentCounts]
+
+        newIndictmentCounts[indictmentCountIndexToUpdate] = {
+          ...newIndictmentCounts[indictmentCountIndexToUpdate],
+          ...update,
+        }
+        return { ...theCase, indictmentCounts: newIndictmentCounts }
+      })
+    },
+    [setWorkingCase],
+  )
+
+  const handleUpdateIndictmentCount = useCallback(
+    async (
+      indictmentCountId: string,
+      updatedIndictmentCount: TIndictmentCount,
```

Comment:
When calling this method, the whole count is passed in. Consider limiting this argument to contain only the changed field(s).